src/adapters/MeetingsJSONAdapter.js

```diff
@@ -384,6 +394,28 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
     return captureStream;
   }
 
+  /**
+   * Returns available media devices.
+   *
+   * @param {string} type String specifying the device type.
+   * @returns {Array} Array containing media devices
+   * @private
+   */
+  // eslint-disable-next-line class-methods-use-this
+  async getAvailableDevices(type) {
+    let devicesByType = null;
+
+    try {
+      devicesByType = await navigator.mediaDevices.enumerateDevices()
+        .filter((device) => device.kind === type);
+    } catch {
```

Comment:
I really don't know why would `enumerateDevices` fail, but if it does, let's not hide the error.

Normally, if there are no devices, the promise is resolved with an empty list.

I would just remove the try block for now, and let the error propagate upwards.

Reply:
filter doesn't wait for the promise to be fulfilled.